libcalico-go/lib/backend/k8s/resources/ipam_affinity.go

```diff
@@ -264,8 +382,10 @@ func (c *blockAffinityClient) List(ctx context.Context, list model.ListInterface
 		}
 		if host == "" || v1kvp.Key.(model.BlockAffinityKey).Host == host {
 			cidr := v1kvp.Key.(model.BlockAffinityKey).CIDR
-			cidr2 := &cidr
-			if requestedIPVersion == 0 || requestedIPVersion == cidr2.Version() {
+			cidrPtr := &cidr
+			// TODO: If we fix this inconsistency between Get and List, we break Watch in that it will not pick up Deleted events anymore.
+			//if (requestedIPVersion == 0 || requestedIPVersion == cidrPtr.Version()) && !v1kvp.Value.(*model.BlockAffinity).Deleted {
+			if requestedIPVersion == 0 || requestedIPVersion == cidrPtr.Version() {
```

Comment:
This is really unfortunate. Since `Watch` depends on this logic (and when we modify for deletion, we set the `Deleted` field), this actually removes any return value from any operation done with `Deleted` set to `"true"`. We might be able to do a hacky fix and make the return values from `Update` and `Delete` return values with the field forced to `"true"` just on the return value (not on what is actually stored), but that just seems wrong to return an object that wasn't what was written.

Reply:
Actually this is fixed by a separate bug in my code. Disregard.